bluemira/base/reactor_config.py

```diff
@@ -201,8 +199,7 @@ def _read_or_return(dict_or_str_path: Union[str, dict]) -> Dict:
             return ReactorConfig._read_json_file(dict_or_str_path)
         elif isinstance(dict_or_str_path, dict):
             return dict_or_str_path
-        else:
-            raise ReactorConfigError("Invalid config_path")
+        raise ReactorConfigError("Invalid config_path")
```

Comment:
It may be nice to support `pathlib.Path` objects here.

Also, this error message is a bit terse. It doesn't really give me any clues about why the path is invalid and what is an acceptable `config_path`.